src/components/TeacherViews/SpecialistUnitListing/SpecialistUnitListing.view.tsx

```diff
@@ -0,0 +1,177 @@
+import { FC, useId } from "react";
+import { useTheme } from "styled-components";
+
+import Flex from "@/components/SharedComponents/Flex";
+import MaxWidth from "@/components/SharedComponents/MaxWidth";
+import UnitList from "@/components/UnitAndLessonLists/UnitList";
+import Grid, { GridArea } from "@/components/SharedComponents/Grid";
+import Box from "@/components/SharedComponents/Box";
+import usePagination from "@/components/SharedComponents/Pagination/usePagination";
+import { Heading, P } from "@/components/SharedComponents/Typography";
+import TabularNav from "@/components/TabularNav";
+import { RESULTS_PER_PAGE } from "@/utils/resultsPerPage";
+import HeaderListing from "@/components/TeacherComponents/HeaderListing/HeaderListing";
+
+type SpecialistPageData = {
+  curriculumData: SpecialistUnitListingData;
+};
+
+export type SpecialistUnitListingData = {
+  units: SpecialistUnit[];
+  developmentalStage: DevelopmentalStage[];
+  programmeSlug: string;
+  subjectSlug: string;
+  subjectTitle: string;
+  themes: Theme[];
+  developmentalStageSlug: string;
+};
+export type DevelopmentalStage = {
+  slug: string;
+  title: string;
+  unitCount: number;
+  lessonCount: number;
+};
+
+export type SpecialistUnit = IndividualSpecialistUnit[];
+
+export type IndividualSpecialistUnit = {
+  slug: string;
+  title: string;
+  nullTitle: string;
+  programmeSlug: string;
+  subjectSlug: string;
+  subjectTitle: string;
+  lessonCount: number;
+  unitStudyOrder: number;
+  expired: boolean;
+  expiredLessonCount: number;
+  themeSlug: string | null;
+  themeTitle: string | null;
+  developmentalStageSlug: string;
```

Comment:
I think developmentalStage needs to be optional (therapies have none, possibly not all specialist does)